### PR TITLE
11 premium route removal

### PR DESCRIPTION
**Pull request by KaiHerrero**
File Additions:
- **premium_routes.csv** --> ted-data/ 
   "updated to include premium routes from the 2023 dataset, this needs to be in this specific filepath for 'limitize-routes.py' to 
     work"
- **limitize-routes.py** --> ted-data/
   "script that creates the -limited gtfs dataset, currently places the limited dataset in the same filepath as itself, can be 
     changed"

Comments:

### CODE REVIEW

**Comment by KaiHerrero** [comment]
**desktop.ini** is an unwanted artifact, feel free to delete

File: ted/gtfs.py [hunk 1]
```diff
@@ -127,62 +127,62 @@ def remove_routes_from_gtfs(gtfs_path: str, output_folder: str, route_ids: list[
...
-#         # Make target output folder with the '-limited' tag
-#         dated_output_path = os.path.join(output_path, curr_dated_entry + '-limited')
-#         os.mkdir(dated_output_path)
-#         zip_entries = os.listdir(os.path.join(gtfs_path, curr_dated_entry))
-#         # Iterate through .zip entries
-#         for curr_zip_entry in zip_entries:
-#             # Find entry zip folder
-#             curr_zip_dir = os.path.join(gtfs_path, curr_dated_entry, curr_zip_entry)
-#             curr_zip_slug = curr_zip_entry.removesuffix('.zip')
+def remove_premeium_routes_from_gtfs(gtfs_path: str, output_path: str, premium_routes_path: str):
```
Comment (wklumpen) [inline ted/gtfs.py:130]: Couple of things:

- Typo in "premium" in the definition
- If `gtfs_path` is a folder, can you name it `gtfs_folder` for consistency?
- Same comment as above but for `output_path`

File: ted/gtfs.py [hunk 2]
```diff
@@ -127,62 +127,62 @@ def remove_routes_from_gtfs(gtfs_path: str, output_folder: str, route_ids: list[
...
+        The path to the folder to remove premium routes from.
+    output_path : str
+        The path to where the new GTFS folder without premium routes will be created.
+    premium_routes_path : str
+        The path to the csv containing the list of premium route slugs and their ids.
+        This must specify a csv file and the csv should be formatted into 'route_slug, route_id' columns
+
+    """
+    premium_routes = pandas.read_csv(premium_routes_path, index_col = False)
+    os.mkdir(output_path)
```
Comment (wklumpen) [inline ted/gtfs.py:145]: Something you can do here is:

``` python
if not os.path.exists(output_folder):
    os.mkdir(output_folder)
```

That way if it already exists it won't throw an error.

File: ted/gtfs.py [hunk 3]
```diff
@@ -127,62 +127,62 @@ def remove_routes_from_gtfs(gtfs_path: str, output_folder: str, route_ids: list[
...
-#             if not(curr_zip_entry.startswith('._')):
-#                 print("Finished parsing: " + curr_zip_entry)
-#         print("\n       ---Finished parsing feed: " + curr_dated_entry + "---\n")
-#     print("Done removing premium routes from specified GTFS folder!")
+            premium_slug_rows = premium_routes.loc[premium_routes['route_slug'] == curr_zip_slug]
+            slug_premium_ids = (premium_slug_rows.iloc[:,1]).tolist()
+
+            #Skip slug labelled __ALL__
+            if '__ALL__' in slug_premium_ids:
+                print(curr_zip_slug + " is a premium route, skipping...")
```
Comment (wklumpen) [inline ted/gtfs.py:168]: Would output `is a premium feed`, since we're skipping the feed entirely (not the route)

File: ted/gtfs.py [hunk 4]
```diff
@@ -127,62 +127,62 @@ def remove_routes_from_gtfs(gtfs_path: str, output_folder: str, route_ids: list[
...
+                        if not os.path.exists(dated_output_path):
+                            os.mkdir(dated_output_path)
+                        copy.write_zip(os.path.join(dated_output_path,curr_zip_entry))
+                except zipfile.BadZipFile:
+                    print(curr_zip_entry, "is not a zipfile, skipping...")
+
+            if not(curr_zip_entry.startswith('._')):
+                print("Finished parsing: " + curr_zip_entry)
+        print("\n       ---Finished parsing feed: " + curr_dated_entry + "---\n")
+    print("Done removing premium routes from specified GTFS folder!")
```
Comment (wklumpen) [inline ted/gtfs.py:185]: You can add the folder name in here for clarity using f-strings:

``` python 
print(f"Done removing premium routes from {gtfs_folder}!")
```

**Review by wklumpen — CHANGES_REQUESTED** [review]

**Review by wklumpen — APPROVED** [review]

**Pull request merged**